src/util/util.js

```diff
@@ -54,20 +60,62 @@ function resolveURLTemplates(URLTemplates, variables = {}, options = {}) {
     if (typeof resolveURL !== 'string') {
       continue;
     }
-
-    for (const key in variables) {
-      const value = variables[key];
-      const macro1 = `[${key}]`;
-      const macro2 = `%%${key}%%`;
-      resolveURL = resolveURL.replace(macro1, value);
-      resolveURL = resolveURL.replace(macro2, value);
-    }
+    resolveURL = replaceUrlMacros(resolveURL, variables);
```

Comment:
if we could avoid mutations it would be great, because it can make debugging very hard if there is a bug.